### PR TITLE
filesystem: add UUID change feature

### PR DESCRIPTION
**Pull request by lszomor**
##### SUMMARY
Add a new feature to the filesystem module for managing UUID.
The feature does not break backward compatibility. It introduces an optional ``uuid`` parameter.

Rationale:
During disk cloning / lvm snapshot creation, the filesystem uuid remains the same. This new feature manages the filesystem UUID in an automated fashion.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
filesystem

##### ADDITIONAL INFORMATION
The ``version_added`` entry in the ``uuid`` parameter's documentation needs to be set.

### CODE REVIEW

File: plugins/modules/filesystem.py [hunk 1]
```diff
@@ -586,6 +647,13 @@ def main():
             out = filesystem.grow(dev)
 
             module.exit_json(changed=True, msg=out)
+        elif same_fs and uuid:
+            if not filesystem.CHANGE_UUID:
+                module.fail_json(changed=False, msg="module does not support UUID change for %s filesystem yet." % fstype)
+
+            out = filesystem.change_uuid(new_uuid=uuid, dev=dev)
+
+            module.exit_json(changed=True, msgout=out)
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:656]:
```suggestion
            module.exit_json(changed=True, msg=out)
```

File: plugins/modules/filesystem.py [hunk 2]
```diff
@@ -79,6 +79,14 @@
     description:
       - List of options to be passed to C(mkfs) command.
     type: str
+  uuid:
+    description:
+      - Set existing filesystem's UUID to the given value.
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:84]: I think this should also work on FS creation.

File: plugins/modules/filesystem.py [hunk 3]
```diff
@@ -79,6 +79,14 @@
...
+      - See xfs_admin(8) (C(xfs)), tune2fs(8) (C(ext2), C(ext3), C(ext4), C(ext4dev)) for possible values.
+      - For C(lvm) the value is ignored, it resets the pv UUID if set.
+      - Supported for C(ext2), C(ext3), C(ext4), C(ext4dev), C(lvm), C(xfs) filesystems.
+      - B(Not idempotent)
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:88]:
```suggestion
      - This is B(not idempotent). Specifying this option will always result in a change.
```

File: plugins/modules/filesystem.py [hunk 4]
```diff
@@ -79,6 +79,14 @@
     description:
       - List of options to be passed to C(mkfs) command.
     type: str
+  uuid:
+    description:
+      - Set existing filesystem's UUID to the given value.
+      - See xfs_admin(8) (C(xfs)), tune2fs(8) (C(ext2), C(ext3), C(ext4), C(ext4dev)) for possible values.
+      - For C(lvm) the value is ignored, it resets the pv UUID if set.
+      - Supported for C(ext2), C(ext3), C(ext4), C(ext4dev), C(lvm), C(xfs) filesystems.
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/filesystem.py:87]: (If you wonder about the syntax, see #6672.)

**Comment by lszomor** [comment]
@felixfontein Thank you for your guidance and fixes!

File: plugins/modules/filesystem.py (hunk 2 above)
Comment (lszomor) [reply plugins/modules/filesystem.py:84]: Good idea, I'm working on the necessary changes.

File: plugins/modules/filesystem.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (lszomor) [reply plugins/modules/filesystem.py]: Thank you! When I saw your PRs regarding the semantic markups, I was thinking about what should I reformat.

File: plugins/modules/filesystem.py [hunk 5]
```diff
@@ -200,13 +239,19 @@ def get_fs_size(self, dev):
         """
         raise NotImplementedError()
 
-    def create(self, opts, dev):
+    def create(self, opts, dev, uuid=None):
         if self.module.check_mode:
             return
 
+        if uuid and self.MKFS_SET_UUID_OPTIONS:
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:246]: Shouldn't this code (and the below) also check whether the filesystem actually supports the `uuid` option (and fail if not, if `uuid` is provided)?

Comment (lszomor) [reply plugins/modules/filesystem.py:246]: It's reasonable to give a feedback, not just ignore the `uuid` silently. Fail feels too harsh, but certainly it draws attention. Thanx!

Comment (felixfontein) [reply plugins/modules/filesystem.py:246]: If the filesystem exists and you try to change the UUID it also fails, so I guess failing here also makes sense.

File: plugins/modules/filesystem.py [hunk 6]
```diff
@@ -525,10 +599,14 @@ def main():
             opts=dict(type='str'),
             force=dict(type='bool', default=False),
             resizefs=dict(type='bool', default=False),
+            uuid=dict(type='str', required=False),
         ),
         required_if=[
             ('state', 'present', ['fstype'])
         ],
+        mutually_exclusive=[
+            ('resizefs', 'uuid')
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:608]:
```suggestion
            ('resizefs', 'uuid'),
```

File: plugins/modules/filesystem.py [hunk 7]
```diff
@@ -576,21 +655,32 @@ def main():
...
+            if resizefs:
+                if not filesystem.GROW:
+                    module.fail_json(changed=False, msg="module does not support resizing %s filesystem yet." % fstype)
+
+                out = filesystem.grow(dev)
+
+                module.exit_json(changed=True, msg=out)
+            elif uuid:
+                if not filesystem.CHANGE_UUID:
+                    module.fail_json(changed=False, msg="module does not support UUID change for %s filesystem yet." % fstype)
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:674]: This shouldn't be necessary, since this is already checked above.
```suggestion
```

**Comment by felixfontein** [comment]
Beside these I think this is safe to merge for 7.1.0.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@lszomor thanks for your contribution!

**Pull request merged**